packages/page-staking/src/Performance/Performance.tsx

```diff
@@ -0,0 +1,321 @@
+          if (currentEraValidatorsValue) {
+            setCurrentEraValidators(currentEraValidatorsValue as CurrentEraValidators);
+          }
+        }).catch(console.error);
+        currentApi.query.elections.committeeSize().then((committeeSizeValue) => {
+          if (committeeSizeValue) {
+            setCommitteeSize(committeeSizeValue as CommitteeSize);
+          }
+        }).catch(console.error);
+      } else {
+        setIsPalletElectionsSupported(false);
+      }
+    }).catch(console.error);
+  }
+
+  useEffect(() => {
+    if (firstBlockInSessionHash) {
+      api.at(firstBlockInSessionHash.toString()).then((result) => {
+        if (result) {
+          getSessionValidators(result);
+        }
+      }).catch(console.error);
+      api.derive.chain.getHeader(firstBlockInSessionHash).then((header) => {
+        if (header && !header.isEmpty && header.author) {
+          setFirstSessionBlockAuthor(header.author.toString());
+        }
+      }).catch(console.error);
+    }
+  },
+  [api, firstBlockInSessionHash]
+  );
+
+  const parseSessionValidatorBlockCount = useCallback((sessionValidatorBlockCountValue: [StorageKey<AnyTuple>, Codec][]) => {
+    const sessionValidatorBlockCountLookup: Record<string, number> = {};
+
+    sessionValidatorBlockCountValue.forEach(([key, values]) => {
+      const account = key.args[0].toString();
+
+      sessionValidatorBlockCountLookup[account] = Number(values.toString());
+
+      if (account === firstSessionBlockAuthor) {
+        // a workaround for the fact that the first session block author is not reflected in that block
+        // elections.sessionValidatorBlockCount state
+        sessionValidatorBlockCountLookup[account] += 1;
+      }
+    });
+    setSessionValidatorBlockCountLookup(sessionValidatorBlockCountLookup);
+  }, [firstSessionBlockAuthor]
+  );
+
+  const setBlockCountLookup = useCallback((currentApi: ApiDecoration<'promise'>) => {
+    currentApi.query.elections.palletVersion().then((version) => {
+      if (Number(version.toString()) >= MINIMUM_SUPPORTED_ELECTIONS_PALLET_VERSION) {
+        currentApi.query.elections.sessionValidatorBlockCount.entries().then((sessionValidatorBlockCountValue) => {
+          if (sessionValidatorBlockCountValue) {
+            parseSessionValidatorBlockCount(sessionValidatorBlockCountValue);
+          }
+        }).catch(console.error);
+      } else {
+        setIsPalletElectionsSupported(false);
+      }
+    }).catch(console.error);
+  }, [parseSessionValidatorBlockCount]
+  );
+
+  useEffect(() => {
+    if (lastBlockInSessionHash && isPalletElectionsSupported) {
+      api.at(lastBlockInSessionHash.toString()).then((result) => {
+        if (result) {
+          setBlockCountLookup(result);
+        }
+      }).catch(console.error);
+    }
+  },
+  [api, lastBlockInSessionHash, isPalletElectionsSupported, setBlockCountLookup]
+  );
+
+  function chooseForSession (validators: AccountId[], count: number, sessionIndex: number) {
```

Comment:
this probably can be replaced with api call to session pallet retrieving the `Validators` storage

Reply:
The current version has the advantage that it works also for future sessions -- and maybe we will need it sometime? (For showing the schedule in the current era.)

Reply:
Turned out that if we use `session.validators` then the code is significantly simplier. If we need a future schedule, we can always add it back.